app/validation/validator.py

```diff
@@ -833,20 +851,34 @@ def _is_contained_in_list(dict_list, key_id):
 
         return False
 
-    def _parse_values(self, schema_json, parsed_key):
-        ignored_keys = ['titles', 'routing_rules', 'skip_conditions']
+    def _parse_values(self, schema_json, parsed_key, path=None):
+        """ generate a list of values with a key of `parsed_key`.
+
+        These values will be returned with the path to them through the object e.g.
+            - (['sections', '[0]', 'groups', '[0]', 'blocks', '[1]', 'question_variants', '[0]', 'question', 'question-2')
```

Comment:
@iwootten has been using json pointers to keep references to specific elements in the schema for the piping and translation work. It would be worth seeing if it's worth using that here.

Reply:
Good point, I considered it early on and then didn't revisit it for some reason. Will try now

Reply:
I've modified this to return JSON pointers. I haven't used a library here, since https://python-json-pointer.readthedocs.io/en/latest/mod-jsonpointer.html doesn't support building paths like this very well.